core/src/Kokkos_Macros.hpp

```diff
@@ -148,12 +149,13 @@
 #define KOKKOS_COMPILER_APPLECC __APPLE_CC__
 #endif
 
-#if defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL)
+#if defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL_LLVM)
 #define KOKKOS_COMPILER_CLANG \
   __clang_major__ * 100 + __clang_minor__ * 10 + __clang_patchlevel__
 #endif
 
-#if !defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL) && defined(__GNUC__)
+#if !defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL) && \
+    !defined(KOKKOS_COMPILER_INTEL_LLVM) && defined(__GNUC__)
```

Comment:
I don't we need to check `INTEL_LLVM` here, since `__clang__` will be defined in that case.

Reply:
I.e. this line can just be reverted.